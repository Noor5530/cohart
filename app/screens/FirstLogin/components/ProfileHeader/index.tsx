import React from "react";
import { Image, Text, View, useWindowDimensions } from "react-native";

import styles from "./style";
interface Props {
  heading: string;
  description: string;
  renderDescription?: React.ReactElement;
}
const Profile: React.FC<Props> = (props: Props) => {
  const { width } = useWindowDimensions();
  const { heading, description } = props;

  return (
    <View style={[styles.container, { width }]}>
      <Image
        style={styles.logo}
        source={require("assets/profileHeaderIcon.png")}
      />
      <View>
        <Text style={styles.header}>{heading}</Text>
        {props.renderDescription ? (
          props.renderDescription
        ) : (
          <Text style={styles.description}>{description}</Text>
        )}
      </View>
    </View>
  );
};

export default Profile;
