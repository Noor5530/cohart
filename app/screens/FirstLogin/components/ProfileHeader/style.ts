import {StyleSheet} from "react-native"

import { appFonts } from 'components/text';
import { primary } from 'config/colors';
import { iconColor } from 'config/colors';
import { heightPercentageToDP as hp, widthPercentageToDP as wp } from "react-native-responsive-screen";

const styles = StyleSheet.create({
    container: {
      height: hp('30'),
      backgroundColor: primary,
      justifyContent: 'flex-end',
      paddingBottom: 30,
    },
    logo: {
      width: wp('22'),
      height: hp('18'),
      position: 'absolute',
      top: 0,
      right: 0
    },
    header: {
      paddingLeft: 38,
      fontFamily: appFonts.InterBold,
      height: 19,
      fontStyle: 'normal',
      fontWeight: 'bold',
      fontSize: hp('2'),
      lineHeight: 19, 
      textTransform: 'uppercase',
      color: iconColor,
    },
    description: {
      fontFamily: appFonts.InterRegular,
      paddingLeft: 38,
      fontStyle: 'normal',
      fontWeight: 'normal',
      fontSize: hp('1.6'),
      textTransform: 'uppercase', 
      color: iconColor,
    },
  });
  
  export default styles