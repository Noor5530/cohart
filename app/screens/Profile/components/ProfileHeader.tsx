import { appFonts } from 'components/text';
import { primary } from 'config/colors';
import React from 'react';
import { Image, StyleSheet, Text, View } from 'react-native';

import { iconColor } from '../../../config/colors';

interface Props {
  heading: string;
  description: string;
}
const Profile: React.FC<Props> = (props: Props) => {
  const { heading, description } = props;

  return (
    <View style={styles.container}>
      <Image
        style={styles.logo}
        source={require('assets/profileHeaderIcon.png')}
      />
      <View>
        <Text style={styles.header}>{heading}</Text>
        <Text style={styles.description}>{description}</Text>
      </View>
    </View>
  );
};

export default Profile;

const styles = StyleSheet.create({
  container: {
    width: '100%',
    height: 216.05,
    backgroundColor: primary,
    justifyContent: 'flex-end',
    paddingBottom: 30,
  },
  logo: {
    width: 88,
    height: 154,
    position: 'absolute',
    bottom: -10,
    right: 0,
  },
  header: {
    paddingLeft: 38,
    fontFamily: appFonts.InterBold,
    height: 19,
    fontStyle: 'normal',
    fontWeight: 'bold',
    fontSize: 15,
    lineHeight: 19,

    textTransform: 'uppercase',

    color: iconColor,
  },
  description: {
    fontFamily: appFonts.InterRegular,
    paddingLeft: 38,
    fontStyle: 'normal',
    fontWeight: 'normal',
    fontSize: 8.28337,
    paddingTop: 12,
    lineHeight: 12 /* identical to box height */,

    textTransform: 'uppercase',

    color: iconColor,
  },
});
