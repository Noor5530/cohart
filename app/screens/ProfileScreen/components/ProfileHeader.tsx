import React, { useState, useEffect } from "react";
import {
  View,
  Text,
  Image,
  StyleSheet,
  TouchableOpacity,
  Linking,
} from "react-native";
import {
  widthPercentageToDP,
  heightPercentageToDP,
} from "react-native-responsive-screen";
import { iconColor, primary } from "../../../config/colors";
import { appFonts } from "../../../components/text";
import Entypo from "react-native-vector-icons/Entypo";
import Fontisto from "react-native-vector-icons/Fontisto";
import AntDesign from "react-native-vector-icons/AntDesign";
import TitleStump from "./TitleStump";
import dummyAvator from "assets/dummyAvator.png";
import { profileHeaderProps } from "../types";
import { useCallback } from "react";
import QrModel from "./QrModel";
import { useNavigation } from "@react-navigation/core";

export default function ProfileHeader(props: profileHeaderProps) {
  const {
    full_name = "",
    username = "",
    title = "",
    location = "",
    instagram = "",
    twitter = "",
    website = "",
    follower_count = 0,
    title_stamp = 1,
    cover_image = null,
    first_name = "",
    _id = "",
  } = props;

  const [userImage, setUserImage] = useState(cover_image ? cover_image : null);
  const [isVisibleQrModal, toggleQrModal] = useState(false);
  const navigation = useNavigation();
  useEffect(() => {
    setUserImage(cover_image ? cover_image : null);
    return () => {
      setUserImage(null);
    };
  }, [cover_image]);
  const openTwitter = useCallback(() => {
    Linking.openURL(`http://twitter.com/${twitter?.trim()}`);
  }, [twitter]);
  const openInstagram = useCallback(() => {
    Linking.openURL(`https://www.instagram.com/${instagram?.trim()}`);
  }, [instagram]);

  const openWeb = useCallback(() => {
    Linking.openURL(website);
  }, [website]);
  const onPressQr = () => {
    toggleQrModal(true);
  };

  function numFormatter(follower_count: any) {
    if (isNaN(follower_count)) {
      return 0;
    } else if (follower_count >= 999 && follower_count < 1000000) {
      return (follower_count / 1000).toFixed(1) + " K"; // convert to K for follower_countber from > 1000 < 1 million
    } else if (follower_count >= 1000000) {
      return (follower_count / 1000000).toFixed(1) + " M"; // convert to M for follower_countber from > 1 million
    } else {
      return follower_count;
    }
  }
  const onClickFans = () => {
    navigation.navigate("FansScreen", {
      _id: _id
    });
  };
  return (
    <View style={styles.container}>
      <View style={styles.leftContainer}>
        {full_name && full_name != "" ? (
          <Text style={styles.name}>{full_name}</Text>
        ) : null}
        {username && username != "" ? (
          <Text style={styles.text}>{"@" + username}</Text>
        ) : null}
        {title && title != "" ? <Text style={styles.text}>{title}</Text> : null}
        {location && location != "" ? (
          <Text style={styles.text}>
            <Entypo
              name="location-pin"
              size={widthPercentageToDP("3.5")}
              color={iconColor}
            />
            {location}
          </Text>
        ) : null}
        {website != "" || instagram != "" || twitter != "" ? (
          <View style={styles.socialContainer}>
            {website && website != "" ? (
              <TouchableOpacity
                style={{ paddingRight: 10 }}
                testID={"openWeb"}
                onPress={openWeb}
              >
                <Fontisto
                  size={widthPercentageToDP(5)}
                  name="world-o"
                  color={iconColor}
                />
              </TouchableOpacity>
            ) : null}
            {twitter && twitter != "" ? (
              <TouchableOpacity
                style={{ paddingRight: 10 }}
                testID={"openTwitter"}
                onPress={openTwitter}
              >
                <AntDesign
                  size={widthPercentageToDP(5)}
                  name="twitter"
                  color={iconColor}
                />
              </TouchableOpacity>
            ) : null}

            {instagram && instagram != "" ? (
              <TouchableOpacity
                style={{ paddingRight: 10 }}
                testID={"openInstagram"}
                onPress={openInstagram}
              >
                <AntDesign
                  size={widthPercentageToDP(5)}
                  name="instagram"
                  color={iconColor}
                />
              </TouchableOpacity>
            ) : null}
          </View>
        ) : null}
        <TouchableOpacity onPress={onClickFans}>
          <Text style={styles.bottomHeading} testID={"numFormat"}>
            {numFormatter(follower_count) + " " + "FANS"}
          </Text>
        </TouchableOpacity>
      </View>
      <View>
        <View style={styles.titleStumpContainer}>
          <TitleStump
            style={styles.titleStump}
            text={
              title_stamp == 1
                ? "PARTNER ORGANIZATON"
                : title_stamp == 2
                  ? "VERIFIED USER"
                  : title_stamp == 3
                    ? "EARLY ADOPTER"
                    : title_stamp == 4
                      ? " COHART-ER"
                      : title_stamp == 5
                        ? " ADVISOR"
                        : title_stamp == 6
                          ? "COCOISTA"
                          : "EARLY ADOPTER"
            }
          />
        </View>
        <View style={styles.imageContainer}>
          <Image
            progressiveRenderingEnabled={true}
            source={userImage ? { uri: userImage } : dummyAvator}
            resizeMode="cover"
            style={styles.imageContainer}
          />
          <TouchableOpacity
            testID={"onPressQr"}
            onPress={onPressQr}
            style={styles.qrCodeView}
          >
            <AntDesign size={22} name="qrcode" color={iconColor} />
          </TouchableOpacity>
        </View>
      </View>
      <QrModel
        isVisible={isVisibleQrModal}
        toggleModal={toggleQrModal}
        name={username}
        first_name={first_name}
        link={"https://web.cohart.co/" + username}
      />
    </View>
  );
}

const styles = StyleSheet.create({
  container: {
    paddingVertical: 22,
    flexDirection: "row",
    justifyContent: "space-between",
    minHeight: heightPercentageToDP(20),
  },
  leftContainer: { width: widthPercentageToDP(40) },
  name: {
    fontSize: widthPercentageToDP(5.1),
    color: iconColor,
    fontFamily: appFonts.InterRegular,
    paddingBottom: 4,
  },
  text: {
    fontSize: widthPercentageToDP(3),
    color: iconColor,
    fontFamily: appFonts.InterRegular,
    fontWeight: "300",
    paddingBottom: 4,
  },
  socialContainer: {
    paddingVertical: 10,
    flexDirection: "row",
    width: widthPercentageToDP(25),
  },
  bottomHeading: {
    fontSize: widthPercentageToDP(3),
    color: iconColor,
    fontFamily: appFonts.InterRegular,
  },
  titleStumpContainer: { top: -25, position: "absolute", right: -70 },
  titleStump: {
    width: widthPercentageToDP(40),
    height: widthPercentageToDP(40),
  },
  imageContainer: {
    width: widthPercentageToDP(25),
    height: widthPercentageToDP(25),
    borderRadius: widthPercentageToDP(13),
  },
  qrCodeView: {
    position: "absolute",
    bottom: 0,
    left: 0,
    backgroundColor: primary,
    width: 40,
    height: 40,
    zIndex: 5,
    borderRadius: 30,
    alignItems: "center",
    justifyContent: "center",
  },
  imageLoading: {
    position: "absolute",
    alignItems: "center",
    justifyContent: "center",
  },
});
