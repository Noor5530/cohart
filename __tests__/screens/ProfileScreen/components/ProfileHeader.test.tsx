import "react-native";
import React from "react";
import ProfileHeader from "../../../../app/screens/ProfileScreen/components/ProfileHeader";
import { render, fireEvent } from "@testing-library/react-native";


jest.useFakeTimers();

test("renders correctly", () => {
    const tree = render(<ProfileHeader />);
    expect(tree).toMatchSnapshot();

});
// test("pickAvatar function with api calling", () => {

//     const { getByTestId } = render(<ProfileHeader />);

//     const mockResponse: AxiosResponse = {
//         data: {
//             data:
//             {
//                 cover_image: "abcd",
//             },

//             // statusCode: 200,
//         },
//         status: 200,
//         statusText: "",
//         headers: undefined,
//         config: {},
//     };
//     const mockResponseForuploadAvatar = jest
//         .spyOn(uploadAvatar, "uploadAvatar")
//         .mockImplementation(() => Promise.resolve(mockResponse));
//     mockResponseForuploadAvatar.mockClear();

//     fireEvent.press(getByTestId("pickAvatar"));

//     expect(getByTestId("pickAvatar")).not.toBeNull();



// });

// test("error message", () => {
//     const { getByTestId } = render(<ProfileHeader />);
//     let openPickerMokeForError = jest.spyOn(ImagePicker, 'openPicker').mockImplementation(() => Promise.reject({ message: "User did not grant library permission." }))

//     fireEvent.press(getByTestId("pickAvatar"));
//     expect(getByTestId("pickAvatar")).not.toBeNull();
//     openPickerMokeForError.mockClear()
// });

// test("catch error ", () => {
//     const { getByTestId } = render(<ProfileHeader />);
//     let openPickerMokeForCatch = jest.spyOn(ImagePicker, 'openPicker').mockImplementation(() => new Error("User cancelled image selection."))

//     fireEvent.press(getByTestId("pickAvatar"));
//     expect(getByTestId("pickAvatar")).not.toBeNull();
//     openPickerMokeForCatch.mockClear()
// });
test("openTwitter function", () => {
    const { getByTestId } = render(<ProfileHeader website="website" twitter="twitter" instagram="instagram" follower_count={isNaN} />);

    fireEvent.changeText(getByTestId("numFormat"));
    expect(getByTestId("numFormat")).not.toBeNull();
});
test("openTwitter function", () => {
    const { getByTestId } = render(<ProfileHeader website="website" twitter="twitter" instagram="instagram" follower_count={10000} />);

    fireEvent.changeText(getByTestId("numFormat"));
    expect(getByTestId("numFormat")).not.toBeNull();
});
test("openTwitter function", () => {
    const { getByTestId } = render(<ProfileHeader website="website" twitter="twitter" instagram="instagram" follower_count={1000000} />);

    fireEvent.changeText(getByTestId("numFormat"));
    expect(getByTestId("numFormat")).not.toBeNull();
});
test("openTwitter function", () => {
    const { getByTestId } = render(<ProfileHeader website="website" twitter="twitter" instagram="instagram" />);

    fireEvent.press(getByTestId("openTwitter"));
    expect(getByTestId("openTwitter")).not.toBeNull();
});
test("openInstagram function", () => {
    const { getByTestId } = render(<ProfileHeader website="website" twitter="twitter" instagram="instagram" />);

    fireEvent.press(getByTestId("openInstagram"));
    expect(getByTestId("openInstagram")).not.toBeNull();
});
test("openWeb function", () => {

    const { getByTestId } = render(<ProfileHeader website="website" twitter="twitter" instagram="instagram" />);

    fireEvent.press(getByTestId("openWeb"));
    expect(getByTestId("openWeb")).not.toBeNull();
});
test("toggleQrModal function", () => {

    const { getByTestId } = render(<ProfileHeader website="website" twitter="twitter" instagram="instagram" />);

    fireEvent.press(getByTestId("onPressQr"));
    expect(getByTestId("onPressQr")).not.toBeNull();
});
