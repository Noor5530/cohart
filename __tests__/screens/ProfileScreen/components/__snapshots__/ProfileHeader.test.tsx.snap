// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "flexDirection": "row",
      "justifyContent": "space-between",
      "minHeight": undefined,
      "paddingVertical": 22,
    }
  }
>
  <View
    style={
      Object {
        "width": undefined,
      }
    }
  >
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#000000",
            "fontFamily": "Inter-Regular",
            "fontSize": undefined,
          }
        }
        testID="numFormat"
      >
        0 FANS
      </Text>
    </View>
  </View>
  <View>
    <View
      style={
        Object {
          "position": "absolute",
          "right": -70,
          "top": -25,
        }
      }
    >
      <View
        style={
          Object {
            "height": undefined,
            "width": undefined,
          }
        }
        text="PARTNER ORGANIZATON"
      >
        <View>
          <RNSVGSvgView
            align="xMidYMid"
            bbHeight="100%"
            bbWidth="100%"
            focusable={false}
            height="100%"
            meetOrSlice={0}
            minX={0}
            minY={0}
            style={
              Array [
                Object {
                  "backgroundColor": "transparent",
                  "borderWidth": 0,
                },
                Object {
                  "height": undefined,
                  "width": undefined,
                },
                Object {
                  "flex": 0,
                  "height": "100%",
                  "width": "100%",
                },
              ]
            }
            text="PARTNER ORGANIZATON"
            vbHeight={200}
            vbWidth={200}
            width="100%"
          >
            <RNSVGGroup>
              <RNSVGPath
                d="M 64,0 A 64,64 0 0 1 -64,0 A 64,64 0 0 1 64,0"
                fill={null}
                matrix={
                  Array [
                    1,
                    0,
                    0,
                    1,
                    100,
                    100,
                  ]
                }
                name="curve"
                propList={
                  Array [
                    "fill",
                    "stroke",
                  ]
                }
                stroke={null}
              />
              <RNSVGText
                content={null}
                dx={Array []}
                dy={Array []}
                fill={4278203383}
                font={
                  Object {
                    "fontFamily": "Inter-Regular",
                    "fontSize": 14.8,
                  }
                }
                matrix={
                  Array [
                    1,
                    0,
                    0,
                    1,
                    100,
                    100,
                  ]
                }
                propList={
                  Array [
                    "fill",
                  ]
                }
                rotate={Array []}
                x={Array []}
                y={Array []}
              >
                <RNSVGTextPath
                  content={null}
                  dx={Array []}
                  dy={Array []}
                  font={Object {}}
                  href="curve"
                  rotate={Array []}
                  startOffset={0}
                  x={Array []}
                  y={Array []}
                >
                  <RNSVGTSpan
                    content="PARTNER ORGANIZATON  ·  PARTNER ORGANIZATON  ·  PARTNER ORGANIZATON  ·  PARTNER ORGANIZATON  ·  PARTNER ORGANIZATON  ·  PARTNER ORGANIZATON  ·  "
                    dx={Array []}
                    dy={Array []}
                    font={Object {}}
                    rotate={Array []}
                    x={Array []}
                    y={Array []}
                  />
                </RNSVGTextPath>
              </RNSVGText>
            </RNSVGGroup>
          </RNSVGSvgView>
        </View>
      </View>
    </View>
    <View
      style={
        Object {
          "borderRadius": undefined,
          "height": undefined,
          "width": undefined,
        }
      }
    >
      <Image
        progressiveRenderingEnabled={true}
        resizeMode="cover"
        source={
          Object {
            "testUri": "../../../app/assets/dummyAvator.png",
          }
        }
        style={
          Object {
            "borderRadius": undefined,
            "height": undefined,
            "width": undefined,
          }
        }
      />
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#E6FF00",
            "borderRadius": 30,
            "bottom": 0,
            "height": 40,
            "justifyContent": "center",
            "left": 0,
            "opacity": 1,
            "position": "absolute",
            "width": 40,
            "zIndex": 5,
          }
        }
        testID="onPressQr"
      >
        <mockedDefaultExport
          color="#000000"
          name="qrcode"
          size={22}
        />
      </View>
    </View>
  </View>
  <mockedDefaultExport
    isVisible={false}
    onBackButtonPress={[Function]}
    onBackdropPress={[Function]}
    onSwipeCancel={[Function]}
    onSwipeMove={[Function]}
    propagateSwipe={true}
    style={
      Object {
        "height": undefined,
      }
    }
    testID="toggleModel"
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "alignSelf": "center",
          "backgroundColor": "#FFFFFF",
          "borderColor": "#000000",
          "borderRadius": 10,
          "borderWidth": 2,
          "justifyContent": "center",
          "paddingHorizontal": undefined,
          "paddingVertical": 25,
          "width": "80%",
        }
      }
    >
      <Text
        style={
          Object {
            "alignSelf": "center",
            "color": "#000000",
            "fontFamily": "Inter-Regular",
            "fontSize": 24,
            "fontWeight": "600",
            "textAlign": "center",
            "width": "80%",
          }
        }
      >
        
      </Text>
      <Text
        style={
          Object {
            "color": "#000000",
            "fontFamily": "Inter-Regular",
            "fontSize": 12,
            "fontWeight": "400",
            "textAlign": "center",
          }
        }
      >
        web.cohart.co/
        
      </Text>
      <View
        style={
          Object {
            "alignSelf": "center",
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "height": undefined,
              "justifyContent": "center",
            }
          }
        >
          <mockedDefaultExport
            color="#000000"
            size="large"
          />
        </View>
      </View>
      <Text
        style={
          Object {
            "color": "#FFFFFF",
            "fontFamily": "Inter-Regular",
            "fontSize": 20,
            "fontWeight": "500",
            "textAlign": "center",
          }
        }
      >
        Share
      </Text>
    </View>
  </mockedDefaultExport>
</View>
`;
